src/ssl.c

```diff
@@ -20484,6 +20530,198 @@ void wolfSSL_sk_GENERAL_NAME_free(WOLFSSL_STACK* sk)
 #endif /* OPENSSL_EXTRA || WOLFSSL_WPAS_SMALL */
 
 #ifdef OPENSSL_EXTRA
+
+/* Creates and returns new DIST_POINT structure */
+WOLFSSL_DIST_POINT* wolfSSL_DIST_POINT_new(void)
+{
+    WOLFSSL_DIST_POINT* dp = NULL;
+    WOLFSSL_DIST_POINT_NAME* dpn = NULL;
+    WOLFSSL_GENERAL_NAMES* gns = NULL;
+
+    WOLFSSL_ENTER("DIST_POINT_new");
+
+    dp = (WOLFSSL_DIST_POINT*)XMALLOC(sizeof(WOLFSSL_DIST_POINT), NULL,
+                                      DYNAMIC_TYPE_OPENSSL);
+    if (dp == NULL) {
+        return NULL;
+    }
+    XMEMSET(dp, 0, sizeof(WOLFSSL_DIST_POINT));
+
+    dpn = (WOLFSSL_DIST_POINT_NAME*)XMALLOC(sizeof(WOLFSSL_DIST_POINT_NAME),
+                                            NULL, DYNAMIC_TYPE_OPENSSL);
+    if (dpn == NULL) {
+        XFREE(dp, NULL, DYNAMIC_TYPE_OPENSSL);
+        return NULL;
+    }
+    XMEMSET(dpn, 0, sizeof(WOLFSSL_DIST_POINT_NAME));
+
+    gns = wolfSSL_sk_new_null();
+    if (gns == NULL) {
+        WOLFSSL_MSG("wolfSSL_sk_new_null error");
+        XFREE(dp, NULL, DYNAMIC_TYPE_OPENSSL);
+        XFREE(dpn, NULL, DYNAMIC_TYPE_OPENSSL);
+        return NULL;
+    }
+    gns->type = STACK_TYPE_GEN_NAME;
+
+    dpn->name.fullname = gns;
+    dpn->type = CRL_DIST_OID;
+    dp->distpoint = dpn;
+
+    return dp;
+}
+
+/* Frees DIST_POINT objects.
+*/
+void wolfSSL_DIST_POINT_free(WOLFSSL_DIST_POINT* dp)
+{
+    WOLFSSL_ENTER("wolfSSL_DIST_POINT_free");
+    if (dp != NULL) {
+        if (dp->distpoint != NULL &&
```

Comment:
I would rewrite as:

```
if (dp->distpoint != NULL) {
    if (dp->distpoint->name.fullname != NULL) {
        wolfSSL_GENERAL_NAMES_free(dp->distpoint->name.fullname);
    }
    XFREE(dp->distpoint, NULL, DYNAMIC_TYPE_OPENSSL);
}
```
That way you aren't checking `dp->distpoint != NULL` twice.